spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -19,25 +19,42 @@
 import sys
 import os
 import logging
-
+import pandas as pd
+import argparse
 import numpy as np
 from matplotlib.ticker import MaxNLocator
 
 from spinalcordtoolbox.aggregate_slicewise import aggregate_per_slice_or_level, save_as_csv, func_wa, func_std, \
-    func_sum, merge_dict
+    func_sum, merge_dict, normalize_csa
 from spinalcordtoolbox.process_seg import compute_shape
 from spinalcordtoolbox.scripts import sct_maths
 from spinalcordtoolbox.csa_pmj import get_slices_for_pmj_distance
 from spinalcordtoolbox.centerline.core import ParamCenterline
 from spinalcordtoolbox.image import add_suffix, splitext
 from spinalcordtoolbox.reports.qc import generate_qc
 from spinalcordtoolbox.utils.shell import SCTArgumentParser, Metavar, ActionCreateFolder, parse_num_list, display_open
-from spinalcordtoolbox.utils.sys import init_sct, set_loglevel
+from spinalcordtoolbox.utils.sys import init_sct, set_loglevel, __sct_dir__
 from spinalcordtoolbox.utils.fs import get_absolute_path
 
 logger = logging.getLogger(__name__)
 
 
+class SeperateNormArgs(argparse.Action):
+    """Seperates predictors from their values, computes interaction between sex and brain-volume and puts the results in a dataframe"""
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        pred = values[::2]
+        val = values[1::2]
+        if len(pred) != len(val):
+            raise parser.error("Values for normalization need to be specified for each predictor.")
+        data_subject = {}
+        for i in range(len(pred)):
+            data_subject[pred[i]] = float(val[i])
```

Comment:
I came up with a slightly shorter way of writing the `list` -> `dict` conversion!

It uses zip (to iterate across both lists at once), and it also uses a dictionary comprehension. :)

```python
        try:
            data_subject = { pred: float(val) for pred, val in zip(values[::2], values[1::2]) }
        except ValueError:
            raise parser.error("Normalization values must be float, and provided in the form: '-normalize predictor1 value1 predictor2 value2 [...]'")
```

I also used try/except because it lets us catch the case where a user provides a non-float value.

Reply:
great! I am testing right now, and the exception doesn't catch the case where the last predictor doesn't have a value:
```
sct_process_segmentation -i t1_seg.nii.gz -normalize brain-volume 960606.0 sex 
```
I am wondering how we can integrate it

Reply:
Ahh, good catch! 

`zip` does have an input argument `strict` to throw an error when the lists are different sizes, but it was [only introduced in Python 3.10](https://stackoverflow.com/a/62773251/7584115). :sweat: 

In that case, we can keep the part you've written:

```python
        pred = values[::2]
        val = values[1::2]
        if len(pred) != len(val):
            raise parser.error("Values for normalization need to be specified for each predictor.")
        try:
            data_subject = { p: float(v) for p, v in pred, val]) }
        except ValueError:
            raise parser.error("Normalization values must be float, and provided in the form: '-normalize predictor1 value1 predictor2 value2 [...]'")
```